.github/workflows/release-prepare.yml

```diff
@@ -50,6 +54,7 @@ jobs:
       # set java and cache
       - name: Set Java and Maven Cache
         uses: actions/setup-java@v4
+        if: ${{ inputs.require_maven }}
```

Comment:
How does java get configuring for the functional tests project?

Reply:
Functional tests doesn't run the any java configuration, it just creates the release tag like the other no maven components

Reply:
Ah yes, as per the comment on the other repo... we are not running the gradle release plugin we simply build the docker image using the Dockerfile and do not issue any gradle commands directly in the workflow.

This may need to change in the future when we automate versioning, and if we split this project into multiple modules then we will likely want to start releasing (and publishing the jar).

Something to bear in mind, as this will require the same features as the maven flow except it will use the gradle plugins and actions etc.

So we'll end up with 3 flavours of release: plain, maven or gradle.